src/org/opensha/commons/calc/magScalingRelations/magScalingRelImpl/TMG2017_MagAreaRel.java

```diff
@@ -0,0 +1,304 @@
+/*******************************************************************************
+ * Copyright 2020 Statement goes here
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.opensha.commons.calc.magScalingRelations.magScalingRelImpl;
+
+
+import org.opensha.commons.calc.magScalingRelations.MagAreaRelationship;
+import org.opensha.commons.util.FaultUtils;
+
+
+/**
+ * <b>Title:</b>TMG2017_MagAreaRel<br>
+ *
+ * <b>Description:</b>  
+ * <p> 
+ * This implements the set of magnitude versus rupture area relation of 
+ * Thingbaijam K.K.S., P.M. Mai and K. Goda (2017, Bull. Seism. Soc. Am., 107, 2225–2246. 
+ * 
+ * Addition to rake, we also use regime (="crustal" or "interface") as an additional parameter 
+ * to distinguish shallow crustal and subduction-interface. The default is "crustal". 
+ * However, the relations for normal faults are also applicable to inslab events.   
+ *  
+ * Notes:
+ * (1) Valid rake is in the range -180 to 180 degrees.  
+ * (2) the standard deviation for area as a function of mag is given for log(area) (base-10) not area.
+ * (2) The current implementation does not handle fixed seismogenic width (or saturated rupture width);
+ *     It will require implementing MagLenRel and MagWidRel as well.   
+ *     
+ * Also see: https://github.com/thingbaijam/sceqsrc
+ *    
+ * </p>
+ *
+ * @version 0.0
+ */
+
+public class TMG2017_MagAreaRel extends MagAreaRelationship {
+    /**
+	 * To Do: Write something here.
+	 */
+	private static final long serialVersionUID = 1L;
+	final static String C = "TMG2017_MagAreaRel";
+    public final static String NAME = "Thingbaijam et al.(2017)";
+    protected String regime = "crustal"; 
+    
+    
+    /**
+     * set regime
+     * @param regime string either "crustal" or "interface"
+     */
+    public void setRegime(String regime) {
+        FaultUtils.assertValidRegime(regime);
+        this.regime = regime;
+      }
+    
+    
+    /**
+     * Computes the median magnitude from rupture area for previously set rake and regime values. 
+     * @param area in km
+     * @return median magnitude MW
+     */
+    public double getMedianMag(double area){
+     
+      // FaultUtils.assertValidRake(rake) does not invalidate NaN - :(   
+      
+      if (Double.isNaN(rake))
+        // return NaN because this estimate is not available.   
+        return  Double.NaN; 
+      
+      if (regime.equalsIgnoreCase("crustal"))
+      		if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+      			// strike slip
+      			return 3.701 + 1.062*Math.log(area)*lnToLog;
```

Comment:
@thingbaijam is it normal practice in these opensha classes to hard-code values like this? Normally tI'd expect these would be constants with meaningful names.

Reply:
Thanks for the comment. This coding style (use of hard-code values to delimit the rake angles and also define the scaling coefficients) is adapted from earlier OpenSHA implementation of scaling relationships - so, I'd say it's quite normal. For anyone with background in scaling relationships, these constants would be meaningful.